### DISCUSSION TITLE
Milvus-Kafka producer configurations

### DISCUSSION THREAD

**Discussion by shaiktas** [milvus-io/milvus#29835] [Q&A and General discussion]
Hello Team, we are trying to set up KafkaConsumer to poll message published by milvus,  We able to poll but could not able to read/parse messages, looks we need to apply decoding or deserialization logic to get this print. Please help us in sharing the exact producer configs so that we can apply similar configs at consumer end to get message parsed correctly. 

i have build a simple kafka consumer to  consume message, below is my consumer code snippet:
---
ev := c.Poll(100)
switch e := ev.(type) {
			case *kafka.Message:
fmt.Printf("%% Message on %s:\n%s\n",
					e.TopicPartition, string(e.Value))
-----

The above code generating below output. 
---
% Message on by-dev-rootcoord-dml_5[0]@111727:
                                                                                                                                                                      ����ǻ�ۙ 
% Message on by-dev-rootcoord-dml_5[0]@111728:
                                                                                                                                                                      ����ǻ�ۙ 
----
Can you pls help here to know print exact message? looks some serialization applied here.. when i tried with below it is erroring out

fmt.Printf("--encoded->>%s",base64.StdEncoding.DecodeString(e.Value))
error with above:
./consumer_example.go:94:65: cannot use e.Value (variable of type []byte) as string value in argument to base64.StdEncoding.DecodeString

**Comment by shaiktas** [comment]
Hello- can some please help here ?? I want to understand what are kafka producer side configuraitons ??

**Comment by yhmo** [comment]
I have no idea.
